ecosystem/sep-0034.md

```diff
@@ -0,0 +1,160 @@
+
+## Token
+
+The JWS can be generated using two JSON objects and a JSON Web Key ([JWK](https://www.rfc-editor.org/rfc/rfc7517.html)). The JWK itself is generated using the wallet signing server's `SIGNING_KEY` and associated secret key listed in the server's [`stellar.toml`](sep-0001.md).
+
+The first JSON object is the [JWS Protected Header](https://www.rfc-editor.org/rfc/rfc7515.html#section-4):
+```json
+{
+    "typ": "JWT",
+    "alg": "EdDSA",
+    "kid": "GCR5WQYXYT4ECBQ3SBALXHICPEVTWKY75XKKZ3ZMF63EXJ5RCWWDO726"
+}
+```
+
+- `typ` is the type of payload encoded
+- `alg` specifies the algorithm used to secure the JWS. 
+  - `EdDSA` stands for Edwards-curve Digital Signature Algorithm and is the same algorithm used for Stellar keypairs. 
+  - Using the same algorithm allows signing servers to use the keypair specified by their [stellar.toml](sep-0001.md)'s `SIGNING_KEY` to sign the JSON-encoded objects described here. 
+- `kid` is the Key Id (stellar public key) used to sign the JWS.
+
+The second JSON object is a JWS Payload. In this case, we'll use a [JWT](http://www.rfc-editor.org/info/rfc7519) containing members similar to the [claims](https://www.rfc-editor.org/rfc/rfc7519.html#section-4.1) described in SEP-10:
+
+```json
+{
+    "iss": "https://walletsigningserver.com",
+    "sub": "GALWLDCKL5FJTBGRB2KXVHNXDQTE422YD7U6EULR7PSZ2OXPTJ3WHTLU",
+    "jti": "aa77983a-e550-4d90-8cc2-d661d7f0b8f6",
+    "kid": "GCR5WQYXYT4ECBQ3SBALXHICPEVTWKY75XKKZ3ZMF63EXJ5RCWWDO726",
+    "aud": "https://anchorserver.com",
+    "exp": "1597789801",
+    "iat": "1597703375"
+}
+```
+
+* `iss` — the fully qualified URL of the home domain for the signing server issuing the token
+* `sub` - the stellar account address of the user requesting the transaction
+* `jti` — the identifier for the transaction being processed by the client and defined by the anchor
+* `kid` - the stellar public key address used to sign the JWS
+* `aud` - the fully qualified URL of the home domain for the anchor server receiving the token
+* `iat` - the [timestamp](https://tools.ietf.org/html/rfc7519#section-4.1.6) for the time this token was issued
+* `exp` - the [timestamp](https://tools.ietf.org/html/rfc7519#section-4.1.4) for the time on or after which the JWT must not be accepted for processing
+
+The two JSON objects should be base64-encoded and concatenated with a `.` character. 
+```
+eyJhbGciOiJFZERTQSJ9
+.
+eyJhdWQiOiJodHRwczovL2FuY2hvcnNlcnZlci5jb20iLCJleHAiOiIxNTk3Nzg5ODAxIiwiaWF0IjoiMTU5NzcwMzM3NSIsImlzcyI6Imh0dHBzOi8vd2FsbGV0c2lnbmluZ3NlcnZlci5jb20iLCJqdGkiOiJhYTc3OTgzYS1lNTUwLTRkOTAtOGNjMi1kNjYxZDdmMGI4ZjYiLCJzdWIiOiJHQ0NYS0NIVkVESDZNR1pZSUhER0lGRDVMSEdEM1JDQVo3TkZMSUFZNk5aQ0ZZQlZTNk1RUEZSNyJ9
+```
+
+The resulting string should then be signed by the server's JSON Web Key (JWK). JWK's can be constructed by using the Stellar public and private keys used by client.
```

Comment:
I think the following code, both the js portion and the python are a bit confusing. 

I suggest not having any code here, just a step by step example process to getting from a stellar keypair to a signed request. Also, if we avoid using random keypairs we can remove statements of the "this *may* look something like" kind.

Once we have SDK support we can retrofit this SEP with proper code examples

Reply:
How would you describe the process of decomposing a stellar keypair into the raw bytes and into the base64 representation? The only way I know how to communicate that process is via an SDK example.

I agree I'd like to not have code in the SEP, but some of these processes are really hard to explain without examples, like how to sign a JWS with a JWK. How do I give steps for this without getting into the RFC details?